specs/core/0_beacon-chain.md

```diff
@@ -1845,6 +1710,140 @@ def process_penalties_and_exits(state: BeaconState) -> None:
 * Let `e = state.slot // EPOCH_LENGTH`. Set `state.latest_penalized_balances[(e+1) % LATEST_PENALIZED_EXIT_LENGTH] = state.latest_penalized_balances[e % LATEST_PENALIZED_EXIT_LENGTH]`
 * Remove any `attestation` in `state.latest_attestations` such that `attestation.data.slot < state.slot - EPOCH_LENGTH`.
```

Comment:
What is the rationale for doing these steps at the beginning of the new epoch and now at the end of the old epoch?